docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+# Default Service Plans in Service Catalog
+
+* [Abstract](#abstract)
+* [Summary](#summary)
+* [Motivation](#motivation)
+* [Goals and Non-Goals](#goals-and-non-goals)
+* [Storage Classes vs. Service Classes](#storage-classes-vs-service-classes)
+* [User Experience](#user-experience)
+  * [Cluster Operator](#cluster-operator)
+     * [Broker Installation](#broker-installation)
+     * [Cluster Customization](#cluster-customization)
+  * [Developer](#developer)
+     * [Service Class Discovery](#service-class-discovery)
+     * [Legacy Provisioning](#legacy-provisioning)
+     * [Provisioning with Defaults](#provisioning-with-defaults)
+     * [Provisioning by Service Type](#provisioning-by-service-type)
+     * [Service Instance Management](#service-instance-management)
+  * [Helm Charts](#helm-charts)
+     * [Broker](#broker)
+     * [Author](#author)
+     * [Consumer](#consumer)
+* [Implementation](#implementation)
+  * [Service Class](#service-class)
+  * [Service Plan](#service-plan)
+  * [Service Instance](#service-instance)
+  * [Service Binding](#service-binding)
+
+## Abstract
+
+Let's wholesale rip off how default storage classes for volumes work and apply
+it to Service Catalog! 😎 This allows us to specify a dependency on a type of
+service without requiring upfront knowledge of the underlying broker.
+
+> "My application requires a mysql database."
+
+> "I just want to try out this application using the recommended defaults."
+
+> "I need to encourage everyone to use the cheapest plans possible in our
+> integration environment by making a small database the default."
+
+## Summary
+
+1.  Add default provisioning and binding parameters to Service Classes.
+1.  Add a "Service Type" field to classes and plans, e.g. mysql or redis.
+1.  Add a is-default-plan annotation to Service Plans to flag it as the default
+    plan for a particular service type.
+1.  Service Instances and Bindings fall back to the defaults set on the class  
```

Comment:
Something that is supported by OSB and is 100% up to the broker's impl is binding types. There could be n types of bindings that come from a result of different parameters being sent to a binding call all resulting in different types of binding results, all of which could be needed as the "default" case.

Reply:
So depending on the binding parameters sent to the same service, I could get back a set of credentials, the results of a log drain, the results of routing changes, etc?

Do you have an example of where a service does this, and how that would play out with defaulting? I am not quite sure I understand the pitfalls that we should be watching out for here and if they would require changes to defaulting or something else.